language/src/rustspec_to_coq.rs

```diff
@@ -741,6 +741,61 @@ fn translate_expression<'a>(e: Expression, top_ctx: &'a TopLevelContext) -> RcDo
     }
 }
 
+fn add_ok_if_result(
```

Comment:
How different this function is from `rustspec_to_cqo_base::add_ok_if_result`? Is there room for factorization?